sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/ExpressionCompiler.java

```diff
@@ -977,12 +998,98 @@ node, new DBSPTypeBool(CalciteObject.EMPTY, false), DBSPOpcode.EQ,
                     arg0 = new DBSPApplyExpression("map", type, arg0.borrow(), cast);
                 }
 
-                String method = "array_append";
+                String method = getCallName(call);
                 if (arg0type.mayBeNull)
                     method += "N";
 
                 return new DBSPApplyExpression(node, method, type, arg0, arg1);
             }
+            case SORT_ARRAY: {
+                if (ops.size() == 1) {
+                    ops.add(1, new DBSPBoolLiteral(true));
+                }
+                DBSPExpression arg0 = ops.get(0);
+                DBSPExpression arg1 = ops.size() == 1 ? new DBSPBoolLiteral(true) : ops.get(1);
+
+                return new DBSPApplyExpression(node, getCallName(call), type, arg0, arg1);
+            }
+            case ARRAY_COMPACT: {
+                DBSPTypeVec vec = type.to(DBSPTypeVec.class);
+                DBSPExpression arg0 = ops.get(0);
+
+                DBSPTypeVec arg0type = arg0.getType().to(DBSPTypeVec.class);
+
+                if (arg0type.sameType(vec)) {
+                    // the expected result type is the same as the argument type
+                    // meaning no element in the array can be nullable
+                    // so just return the current argument as is
+                    return arg0;
+                }
+
+                String method = "array_compact";
+                if (arg0type.mayBeNull)
+                    method += "N";
+
+                return new DBSPApplyExpression(node, method, type, arg0);
+            }
+            case ARRAY_MAX:
+            case ARRAY_MIN:
+            {
+                DBSPExpression arg0 = ops.get(0);
+                DBSPTypeVec vec = arg0.getType().to(DBSPTypeVec.class);
+                DBSPType elemType = vec.getElementType();
+
+                String method = getCallName(call);
+                if (elemType.mayBeNull)
+                    method += "N";
+
+                return new DBSPApplyExpression(node, method, type, arg0);
+            }
+            case ARRAY_REVERSE:
```

Comment:
can this be unified with the previous case?
In general, if many of these are similar consider creating a helper function.

Reply:
I think this is cleaner. The previous case checks the nullability of elements in the array while this case checks the nullability of the first argument (the array itself).